### DISCUSSION TITLE
Minor improvements/suggestions

### DISCUSSION THREAD

**Discussion by Egezenn** [oxdc/obsidian-vertical-tabs#62] [Ideas]
Hello, I'd like to say that I love the plugin! It replaced the workspaces for me (it's tiresome compared to this plugin because you've to name the workspace, save the workspace, load the workspace, use a modal popup to manage all this.. yada yada yada <sub>i hate modals</sub>). Here are my honest thoughts about some things that I think would benefit us all:

- Sorting of groups themselves.
- Remembering collapsed and expanded groups.
- The elements (mainly the group itself and the show/hide button for it) aren't accesible/interactable via `Tab` or arrow keys while the plugin sidebar has focus. Which might be a niche use case but is a thing I like to do to not reach for my mouse in file explorer of Obsidian. Would love to see it in the plugin too!
- Backgrounds to `Show/Hide` button. I ended up including some CSS in my snippet for it, which *I know* it looks out of place, but it's a whole lot better than just light gray icons which are indistinguishable to my eye. Could be an option in the settings of the plugin.

<img title="before" alt="after" align="right" src="https://github.com/user-attachments/assets/5258abf8-8a39-4a36-9c2d-278fdc6b0da9">
<img title="after" alt="before" align="right" src="https://github.com/user-attachments/assets/29a7171e-96eb-42c1-8256-e37afb101dce">

```
.clickable-icon.action-button.action-eye>svg {
    color: #000;
    background-color: #F00;
    border-radius: 8px;
  }

.clickable-icon.action-button.action-eye-off>svg {
    color: #000;
    background-color: #0F0;
    border-radius: 8px;
  }
```

- Splits inside groups. This might be asking for too much since it contradicts what the plugin aims for since it's not a dropdown replacement for workspaces. *Or is it*?

**Comment by oxdc** [comment]
Hi! Thank you for your suggestions! I'll convert all the feature requests to discussions and close this issue. We may discuss the details there. It would be more easy to keep track with GitHub's discussion feature.

**Comment by oxdc** [comment]
> Sorting of groups themselves.
> Remembering collapsed and expanded groups.
> Splits inside groups.

I'm working on these features. It will take some time though, as there would be a huge amount of work. Please be patient. 😊

> aren't accessible/interactable via Tab or arrow keys
> light gray icons which are indistinguishable to my eye

Whoa, I haven't notice these accessibility issues! Thank you for reminding me. I don't have much experience on these things. I'll make some improvements but I might not able to deviate too much from Obsidian's UI & UX design in order to provide a consistent experience for users. I'll try my best though.

**Reply by Egezenn** [reply]
> I'm working on these features. It will take some time though, as there would be a huge amount of work. Please be patient. 😊

Glad to hear! Will be thrilled to see them in action :)

> Whoa, I haven't notice these accessibility issues! Thank you for reminding me.

Just happy to be of help! My sincere thanks for creating such an amazing plugin.

> I'll make some improvements but I might not able to deviate too much from Obsidian's UI & UX design in order to provide a consistent experience for users.

The background is probably an only me thing, because I removed most of the spacing in the snippet, so it was really hard to see when all the elements were cramped :D
However wouldn't it be better to change the state of the icons to represent the state they're in rather than the state they'll be on when you interact with them? I'm not too experienced with UX but it was a bit confusing at first.

**Comment by oxdc** [comment]
> Splits inside groups. This might be asking for too much since it contradicts what the plugin aims for since it's not a dropdown replacement for workspaces.

For this FR, specifically, I want to implement a feature called "Sections", which allows you to make headings/subgroups/sections inside a group! In the vertical tabs, it should look something like this,

<img width="275" alt="image" src="https://github.com/user-attachments/assets/ca271e3c-5eb5-4d64-a444-eceea115dce0">

I copied this image from the ads of Things 3. Remember these are sections inside a group.

For the horizontal tabs, I think the best would be something like Chrome's tab group, a highlighted tag for each section,

<img width="827" alt="image" src="https://github.com/user-attachments/assets/5379af5f-3fe1-4610-9d75-c444408df4bf">

The trouble is, I haven't figured out a way to make this work. I am still working on it.

**Reply by Egezenn** [reply]
Wow, great idea!
It's the kind of thing that I'd expect to be complicated to implement, Will be eager to see it working!

**Comment by kmaustral** [comment]
I still find a place for Workspaces with Vertical Tabs when I want a major context shift. But VT is great when you have a cluster of different projects on your bench at one time. But I do find that sometimes clicking a link in a note in one group leads to the new note opened in a different group.

**Reply by oxdc** [reply]
> I still find a place for Workspaces with Vertical Tabs when I want a major context shift. But VT is great when you have a cluster of different projects on your bench at one time.

Thank you! I'll take this into consideration.

> But I do find that sometimes clicking a link in a note in one group leads to the new note opened in a different group.

This is a problem of Obsidian, see the discussion and solution here: #49
